core-feature-pack/src/main/resources/content/bin/standalone.bat

```diff
@@ -233,14 +236,26 @@ if not "%PRESERVE_JAVA_OPTS%" == "true" (
         move /y "%JBOSS_LOG_DIR%\gc.log.4" "%JBOSS_LOG_DIR%\backupgc.log.4" > nul 2>&1
         move /y "%JBOSS_LOG_DIR%\gc.log.*.current" "%JBOSS_LOG_DIR%\backupgc.log.current" > nul 2>&1
 
-        "%JAVA%" -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=3M -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:-TraceClassUnloading -version > nul 2>&1
+        setlocal EnableDelayedExpansion
+        if "x!JAVA_9_PLUS!" == "x" (
+            "%JAVA%" -verbose:gc -Xlog:gc*:file="%JBOSS_LOG_DIR%\gc.log":time,uptimemillis:filecount=5,filesize=3M -version > nul 2>&1
+        ) else (
+            "%JAVA%" -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=3M -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:-TraceClassUnloading -version > nul 2>&1
+        )
+
         if not errorlevel == 1 (
           if not exist "%JBOSS_LOG_DIR" > nul 2>&1 (
             mkdir "%JBOSS_LOG_DIR%"
           )
-		set JAVA_OPTS=%JAVA_OPTS% -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:GCLogFileSize=3M -XX:-TraceClassUnloading
+            if "x!JAVA_9_PLUS!" == "x" (
+                set JAVA_OPTS=%JAVA_OPTS% -verbose:gc -Xlog:gc*:file="%JBOSS_LOG_DIR%\gc.log":time,uptimemillis:filecount=5,filesize=3M
+            } else {
+                set JAVA_OPTS=%JAVA_OPTS% -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:GCLogFileSize=3M -XX:-TraceClassUnloading
+            )
         )
        )
+
+       setlocal DisableDelayedExpansion
```

Comment:
This didn't work for me on Windows 7. I had to change it to something like
```
        if "x%JAVA_9_PLUS%" == "x" (
            set JAVA_OPTS=%JAVA_OPTS% -Xlog:gc*:file=\"%JBOSS_LOG_DIR%\gc.log\":time,uptimemillis:filecount=5,filesize=3M
        ) else (
            "%JAVA%" -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=3M -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:-TraceClassUnloading -version > nul 2>&1
            if not errorlevel == 1 (
                set JAVA_OPTS=%JAVA_OPTS% -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -Xloggc:"%JBOSS_LOG_DIR%\gc.log" -XX:GCLogFileSize=3M -XX:-TraceClassUnloading
            )
        )
       )
```

Note I'm not using delayed expansion and the quotes need to be escaped `\"`. Also there's no need to check that `-Xlog` is supported by the JVM.

Reply:
On EAP side we care only about Windows Server, but understand the concern for community users and developers. Still, Windows 7 is pretty old :)

Reply:
It's the only version of Windows I have to test on in a VM :) I do agree it's rather old at this point though. It was the escaping of the quote that didn't work for me.